### PR TITLE
Moving Cosmos Exception child classes to implementation package

### PR DESCRIPTION
**Pull request by kushagraThapar**
* In this PR -> I have experimented with Java 11 - module-info.java file - exporting only com.azure.cosmos package. 
* When exceptions are in implementation package -> they are working just fine. 
* End user of course will not be able to explicitly use them in their code, since exceptions will be in implementation package. 
* Tested with ConflictException, NotFoundException. Exception is logging just fine.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java [hunk 1]
```diff
@@ -3,14 +3,14 @@
 
 package com.azure.cosmos.implementation.directconnectivity;
```
Comment (David-Noble-at-work) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java]: Please consider sorting the imports here as one of them may now be out of order.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelper.java [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 package com.azure.cosmos.implementation.directconnectivity;
```
Comment (David-Noble-at-work) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelper.java]: Please consider sorting the imports here as one of them may now be out of order.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java [hunk 3]
```diff
@@ -6,9 +6,9 @@
 import com.azure.cosmos.BridgeInternal;
```
Comment (David-Noble-at-work) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java]: Please consider sorting the imports here as one of them may now be out of order.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java [hunk 4]
```diff
@@ -5,7 +5,7 @@
```
Comment (David-Noble-at-work) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java]: Please consider sorting the imports here as one of them may now be out of order.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java [hunk 5]
```diff
@@ -5,12 +5,12 @@
```
Comment (David-Noble-at-work) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java]: Please consider sorting the imports here as one of them may now be out of order.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java]: Done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java (hunk 3 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java]: Done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java (hunk 4 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java]: Done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java (hunk 5 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java]: Done

**Pull request merged**